packages/core/src/domain/session/sessionStoreManager.ts

```diff
@@ -0,0 +1,256 @@
+import { clearInterval, setInterval, setTimeout } from '../../tools/timer'
+import { Observable } from '../../tools/observable'
+import { dateNow } from '../../tools/utils/timeUtils'
+import { throttle } from '../../tools/utils/functionUtils'
+import { generateUUID } from '../../tools/utils/stringUtils'
+import { SESSION_EXPIRATION_DELAY, SESSION_TIME_OUT_DELAY } from './sessionConstants'
+import { initCookieStore } from './sessionCookieStore'
+import type { SessionState, SessionStore, StoreInitOptions } from './sessionStore'
+import { isSessionInExpiredState } from './sessionStore'
+
+export interface SessionStoreManager {
+  expandOrRenewSession: () => void
+  expandSession: () => void
+  getSession: () => SessionState
+  renewObservable: Observable<void>
+  expireObservable: Observable<void>
+  expire: () => void
+  stop: () => void
+}
+
+/**
+ * Different session concepts:
+ * - tracked, the session has an id and is updated along the user navigation
+ * - not tracked, the session does not have an id but it is updated along the user navigation
+ * - inactive, no session in store or session expired, waiting for a renew session
+ */
+export function startSessionStoreManager<TrackingType extends string>(
+  options: StoreInitOptions,
+  productKey: string,
+  computeSessionState: (rawTrackingType?: string) => { trackingType: TrackingType; isTracked: boolean }
+): SessionStoreManager {
+  const renewObservable = new Observable<void>()
+  const expireObservable = new Observable<void>()
+
+  const sessionStore = initCookieStore(options)
```

Comment:
I do agree that a strategy pattern is a good option!
About the naming, I found it hard to identified which of sessionStore, sessionInCookie and sessionInLocalStorage implement the strategy. 
Maybe since all these files are located in session/ folder we could do something like this:
- sessionManager (sry for putting it back... I don't have better name to make clear that it is this file uses the storage)
- storageStrategy
- cookieStorageStrategy
- localStorageStrategy

Reply:
fyi, there is already a sessionManager 😅

Reply:
Done in https://github.com/DataDog/browser-sdk/pull/2261/commits/305deaf969999abdf917486304b13e59ebd8eda7